### PR TITLE
Install TA 2017 offline

### PR DESCRIPTION
**Pull request by nigurr**

### CODE REVIEW

File: Tasks/DeployVisualStudioTestAgent/InstallTestAgent.ps1 [hunk 1]
```diff
@@ -10,6 +29,11 @@ function InstallTestAgent2017
         throw "Test agent source path '{0}' is not accessible to the test machine. Please check if the file exists and that test machine has access to that machine" -f $SetupPath
     }
 
+    # First we need to install the certificates for TA 2017
+    $SetupDir = Split-Path -Path $SetupPath
+    Write-Verbose "Installing test agent certificates"
+    Get-ChildItem -Path "$SetupDir\certificates\*.p12" -ErrorAction SilentlyContinue | Import-PfxCertificate -CertStoreLocation Cert:\CurrentUser\My -Exportable
```
Comment (acesiddhu) [inline Tasks/DeployVisualStudioTestAgent/InstallTestAgent.ps1]: >Import-PfxCertificate [](start = 87, length = 21)

not there iin win 7 or 2k8r2

File: Tasks/DeployVisualStudioTestAgent/InstallTestAgent.ps1 [hunk 2]
```diff
@@ -20,27 +44,29 @@ function InstallTestAgent2017
...
-    # Fish $p.WaitForExit can't be run on remote machines. Wasted time 4hrs. Should have read MSDN
-    do {
-        $waitProcess = Get-Process -Id $($p.Id) -ErrorAction SilentlyContinue
-        Start-Sleep -Seconds 30
-        Write-Verbose -Verbose "Waiting for installtion to finish"
-    } while($waitProcess -and $waitProcess.Name -ilike "*testagent*")
-    
-    return 0
+
+    $p.WaitForExit(1500000) # it shouldn't take more than 25mins. so set time out
```
Comment (acesiddhu) [inline Tasks/DeployVisualStudioTestAgent/InstallTestAgent.ps1]: >1500000 [](start = 19, length = 7)

1 hour as build timeout is 1 hour

File: Tasks/DeployVisualStudioTestAgent/InstallTestAgent.ps1 [hunk 3]
```diff
@@ -1,3 +1,22 @@
...
+        }
+        Unzip $SetupArchive $Destination
+    }
+    Catch {
+        Write-Warning $_
+    }
+}
+
+function FindAgentSetupExe ($SetupPath) {
+    return (Get-ChildItem -Filter "*testagent.exe" -Path $SetupPath -ErrorAction SilentlyContinue | Select-Object -First 1).FullName
```
Comment (RanjanarMS) [inline Tasks/DeployVisualStudioTestAgent/InstallTestAgent.ps1]: If multiple error out...

Comment (RanjanarMS) [inline Tasks/DeployVisualStudioTestAgent/InstallTestAgent.ps1]: Or remove the find

File: Tasks/DeployVisualStudioTestAgent/InstallTestAgent.ps1 (hunk 2 above)
Comment (nigurr) [inline Tasks/DeployVisualStudioTestAgent/InstallTestAgent.ps1]: 1 hr.. Log..

**Pull request merged**